From 9fd4d7ce4f365b6c39f6b2d01bb610bc0f80df22 Mon Sep 17 00:00:00 2001
From: Ping Gao <ping.a.gao@intel.com>
Date: Wed, 6 Sep 2017 19:57:57 +0800
Subject: [PATCH 0943/2328] drm/i915/gvt: Add a module parameter to disable
 conformance check

By default the guest conformance check is enabled.

Signed-off-by: Ping Gao <ping.a.gao@intel.com>
Reviewed-by: Kevin Tian <kevin.tian@intel.com>
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>

Conflicts:
	drivers/gpu/drm/i915/i915_params.c
	drivers/gpu/drm/i915/i915_params.h

Change-Id: I3c3357733818e4529a0dd0d725b38c5d739bef62
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/gvt/scheduler.c | 3 ++-
 drivers/gpu/drm/i915/i915_params.c   | 4 ++++
 drivers/gpu/drm/i915/i915_params.h   | 3 ++-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/scheduler.c b/drivers/gpu/drm/i915/gvt/scheduler.c
index f6a1459..5a0fb7e 100644
--- a/drivers/gpu/drm/i915/gvt/scheduler.c
+++ b/drivers/gpu/drm/i915/gvt/scheduler.c
@@ -266,7 +266,8 @@ static int dispatch_workload(struct intel_vgpu_workload *workload)
 
 	ret = intel_gvt_scan_and_shadow_workload(workload);
 
-	if (intel_gvt_vgpu_conformance_check(vgpu, ring_id))
+	if (i915.enable_conformance_check
+			&& intel_gvt_vgpu_conformance_check(vgpu, ring_id))
 		gvt_err("vgpu%d unconformance guest detected\n", vgpu->id);
 
 	if (ret)
diff --git a/drivers/gpu/drm/i915/i915_params.c b/drivers/gpu/drm/i915/i915_params.c
index d7b0c6a..1632f34 100644
--- a/drivers/gpu/drm/i915/i915_params.c
+++ b/drivers/gpu/drm/i915/i915_params.c
@@ -67,6 +67,7 @@ struct i915_params i915 __read_mostly = {
 	.enable_gvt = false,
 	.enable_pvmmio = 1,
 	.enable_gvt_oos = 1,
+	.enable_conformance_check = true,
 };
 
 module_param_named(modeset, i915.modeset, int, 0400);
@@ -267,3 +268,6 @@ MODULE_PARM_DESC(enable_pvmmio,
 
 module_param_named(enable_gvt_oos, i915.enable_gvt_oos, bool, 0400);
 MODULE_PARM_DESC(enable_gvt_oos, "To toggle the gvt ppgtt page table OOS (Out of Sync) feature.");
+
+module_param_named(enable_conformance_check, i915.enable_conformance_check, bool, 0400);
+MODULE_PARM_DESC(enable_conformance_check, "To toggle the GVT guest conformance feature.");
diff --git a/drivers/gpu/drm/i915/i915_params.h b/drivers/gpu/drm/i915/i915_params.h
index 712a27c..33d2fe3 100644
--- a/drivers/gpu/drm/i915/i915_params.h
+++ b/drivers/gpu/drm/i915/i915_params.h
@@ -70,7 +70,8 @@
 	func(bool, enable_dpcd_backlight); \
 	func(bool, enable_gvt); \
 	func(bool, enable_pvmmio); \
-	func(bool, enable_gvt_oos)
+	func(bool, enable_gvt_oos); \
+	func(bool, enable_conformance_check)
 
 #define MEMBER(T, member) T member
 struct i915_params {
-- 
2.7.4

